Lib/pdb.py

```diff
@@ -1728,8 +1728,11 @@ def main():
             print("Running 'cont' or 'step' will restart the program")
             t = sys.exc_info()[2]
             pdb.interaction(None, t)
-            print("Post mortem debugger finished. The " + mainpyfile +
-                  " will be restarted")
+            if pdb._user_requested_quit:
+                break
```

Comment:
After experimenting, I understand the motivation for this change here, but I also worry that it might affect other workflows we haven't considered. Could there be other use-cases where a different Exception occurred and `pdb._user_requested_quit` is True, but we would want that to fall through (not break)?

Reply:
Good question. self._user_requested_quit will be True only when we get quit()/exit() or EOF. ISTM that it's safe to exit in both cases. 

I'm pretty sure I've had to kill pdb processes in the past, when they stopped responding to quit(), so I would guess it's more likely that there are other code paths that would benefit from this change.